### DISCUSSION TITLE
Tell Pocketbase to pick up a file already at the storage location without going through the API

### DISCUSSION THREAD

**Discussion by thangngoc89** [pocketbase/pocketbase#4858] [Q&A] [answered]
I'm using Pocketbase to build a new PoC product and it has been wonderful so far for a solo developer. My app needs to handle large file uploading (100MB-500MB) and I would like to add resumable uploading capability to the frontend. 

My plan is using [tusd](https://github.com/tus/tusd) + uppy for this function. They seems to be well-tested, have wide-spread support and backed by a big company who are actively developing and dogfooding.

Planned implementation workflow: `tusd` or other upload mechanism would place the file at the correct path in S3 position: `/collectionId/recordId/file_name_random10chars.ext`

I've read your reply on [another discussion thread](https://github.com/pocketbase/pocketbase/discussions/4851#discussioncomment-9288204) on how to programmatically adding a file but as I understand it, this mean downloading and reuploading to the same location. Given that I have the ability to directly save the file to the S3 location, I would like to somehow tell Pocketbase that the file is already present,

I've gone with the route of extending Pocketbase using golang.

**Comment by ganigeorgiev** [comment, marked as answer]
>  Given that I have the ability to directly save the file to the S3 location, I would like to somehow tell Pocketbase that the file is already present

In the database we store only the record file name. If you have uploaded the file externally in `/collectionId/recordId/file_name_random10chars.ext` location then in this case you can directly set the name of the file as your record field value.

```js
record.Set("yourFileField", "file_name_random10chars.ext")
app.Dao().SaveRecord(record)
```

**Reply by thangngoc89** [reply]
Wow. Thanks for you answer. I couldn't believe it's this simple.
